### PR TITLE
fixes for initialization

### LINKED ISSUE #8: Trying to add opentracing support to an app without specifying a tracer fails
Desire: We can add tracing support to an app without being required to configure a tracer.

Currently: It breaks the app.

```
Starting development server at http://127.0.0.1:8000/
Quit the server with CONTROL-C.
Internal Server Error: /
Traceback (most recent call last):
  File "/home/m/.virtualenvs/quartz/local/lib/python2.7/site-packages/django/core/handlers/base.py", line 204, in get_response
    response = middleware_method(request, response)
  File "/home/m/.virtualenvs/quartz/local/lib/python2.7/site-packages/django_opentracing/middleware.py", line 40, in process_response
    self._tracer._finish_tracing(request)
AttributeError: 'Tracer' object has no attribute '_finish_tracing'
quartz[(process)d]: [django.request][ERROR] Internal Server Error: /
Traceback (most recent call last):
  File "/home/m/.virtualenvs/quartz/local/lib/python2.7/site-packages/django/core/handlers/base.py", line 204, in get_response
    response = middleware_method(request, response)
  File "/home/m/.virtualenvs/quartz/local/lib/python2.7/site-packages/django_opentracing/middleware.py", line 40, in process_response
    self._tracer._finish_tracing(request)
AttributeError: 'Tracer' object has no attribute '_finish_tracing'
```

PR to follow

### PR DESCRIPTION
**Pull request by mikebryant**

### CODE REVIEW

**Comment by mikebryant** [comment]
(Where'd Travis come from? I don't see a `.travis.yml`... Going to ignore it!)

**Comment by mikebryant** [comment]
Here's how I've now configured my test app for Jaeger:
```
settings.py:
MIDDLEWARE_CLASSES = [
    'django_opentracing.OpenTracingMiddleware',
...
]
OPENTRACING_TRACE_ALL = True
OPENTRACING_TRACER_CALLABLE = 'example_project.local_settings.tracer'

example_project/local_settings.py:
def tracer():
    from jaeger_client import Config
    return Config(config={}, service_name='test').initialize_tracer()
```

File: django_opentracing/tracer.py [hunk 1]
```diff
@@ -63,12 +65,12 @@ def _apply_tracing(self, request, view_func, attributes):
         span = None
         operation_name = view_func.__name__
         try:
-            span_ctx = self._tracer.extract(opentracing.Format.HTTP_HEADERS, headers)
-            span = self._tracer.start_span(operation_name=operation_name, child_of=span_ctx)
+            span_ctx = opentracing.tracer.extract(opentracing.Format.HTTP_HEADERS, headers)
```
Comment (bhs) [inline django_opentracing/tracer.py]: This surprises me... usually the pattern is to *default* to the `opentracing.tracer` but to allow for overrides (a la the `tracer` param to `__init__`). That way it remains possible to use the django opentracing instrumentation without *requiring* the use of the global tracer.

Comment (mikebryant) [reply django_opentracing/tracer.py]: That's a fair point. My rationale here was to allow for lazy initialization, which requires a global object somewhere..

I'll change this so you can use the instrumentation without the global tracer. (But obviously then the user needs to deal with lazy initialization themself)

Thanks for the input :)

Comment (mikebryant) [reply django_opentracing/tracer.py]: @bhs How's that?

File: django_opentracing/tracer.py [hunk 2]
```diff
@@ -16,6 +20,13 @@ def __init__(self, tracer):
         else:
             self._trace_all = True
 
+    @property
+    def _tracer(self):
```
Comment (bhs) [inline django_opentracing/tracer.py]: great, thanks!

**Comment by mikebryant** [comment]
@bhs Not waiting for anyone else.

I've tested it with my example project with Jaeger, and it seems fine.

I've just made a minor change to the docs given #11 to make it clear that the example parameters are just that

**Comment by bhs** [comment]
Great -- keys do that then merge this. Thanks!

**Comment by mikebryant** [comment]
@bhs Sorry, I meant I have done that :)
It's ready for merge

**Comment by zkanda** [comment]
I tested it and it seems to work fine, what else do we need to get this merge and release?

**Comment by bhs** [comment]
@mikebryant @zkanda apologies for the delay – my overall github notifications situation is very grim! Merged. I will attempt a release now (assuming the scripts still work).

**Comment by bhs** [comment]
@mikebryant @zkanda would either of you be willing to become (co-)owners on pypi for this? The setup.py is apparently out of date in some way(*) and thus I can't publish this artifact using the script here (which I did not write).

Let me know. At the moment the release is blocked on finding someone who understands pypi well enough to fix this (probably really trivial).

(*): I really should not be maintaining this as I (1) am far from a python expert, and (2) am struggling to keep up with my various "notification streams", meaning that simple improvements like this take way too long to merge.

**Comment by mikebryant** [comment]
@bhs I would be happy to help maintain this.
My Pypi account is `LeaChim`

I'm also happy to get this setup with continuous integration, Travis etc, to enable automated releases and testing. (might also need github access to change Web hook settings for that)

**Comment by bhs** [comment]
@mikebryant that's awesome. Unfortunately I am not an `owner` on the django-OT package... I've emailed Kathy Camenzind who *is* an owner there and will grant you privs once she gets back to me.

In the meantime I've made you an owner on this github repo... at least I can control that part. :)

**Comment by bhs** [comment]
@mikebryant all done:

image

**Comment by mikebryant** [comment]
See #12

**Pull request merged**